lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -678,9 +750,12 @@ PromotedParamCloner::populateCloned() {
       OrigPromotedParameters.insert(*I);
 
       NewPromotedArgs[ArgNo] = promotedArg;
-
-      // All uses of the promoted box should either be projections, which are
-      // folded when visited, or copy/destroy operations which are ignored.
+      // We only promote boxes used in apply or projections or copy/destroy
+      // value operations.
+      // We should never see an apply user of the box, because we rewrite the
+      // applies and specialize the callees in dfs order.
```

Comment:
Bravo! I love the way that you are telling now the "how", but the "why", the larger context!